jetbrains-core/tst/software/aws/toolkits/jetbrains/services/telemetry/TelemetryServiceTest.kt

```diff
@@ -18,52 +18,49 @@ import org.junit.Test
 import software.aws.toolkits.core.region.AwsRegion
 import software.aws.toolkits.core.telemetry.DefaultMetricEvent.Companion.METADATA_NA
 import software.aws.toolkits.core.telemetry.DefaultMetricEvent.Companion.METADATA_NOT_SET
-import software.aws.toolkits.core.telemetry.DefaultTelemetryBatcher
 import software.aws.toolkits.core.telemetry.MetricEvent
+import software.aws.toolkits.core.telemetry.TelemetryBatcher
 import software.aws.toolkits.jetbrains.core.MockResourceCache
 import software.aws.toolkits.jetbrains.core.credentials.MockCredentialsManager
 import software.aws.toolkits.jetbrains.core.credentials.MockProjectAccountSettingsManager
 import software.aws.toolkits.jetbrains.core.region.MockRegionProvider
-import software.aws.toolkits.jetbrains.settings.MockAwsSettings
-import java.util.UUID
+import software.aws.toolkits.jetbrains.settings.AwsSettings
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 class TelemetryServiceTest {
-    private val batcher = mock<DefaultTelemetryBatcher> {
-        on { enqueue(any<MetricEvent>()) }.then {
-            mock.enqueue(listOf(it.getArgument<MetricEvent>(0)))
-            null
-        }
-    }
+    private class TestTelemetryService(batcher: TelemetryBatcher) : TelemetryService(NoOpPublisher(), batcher)
 
     @Rule
     @JvmField
     val projectRule = ProjectRule()
 
     @After
     fun tearDown() {
+        AwsSettings.getInstance().isTelemetryEnabled = false
+
         MockProjectAccountSettingsManager.getInstance(projectRule.project).reset()
         MockCredentialsManager.getInstance().reset()
         MockRegionProvider.getInstance().reset()
     }
 
     @Test
-    fun testInitialChangeEvent() {
+    fun testInitialTelemetrySetting() {
+        AwsSettings.getInstance().isTelemetryEnabled = true
```

Comment:
Should we put this in a @before, or does that break other tests?

Reply:
Doesn't seem to. Nothing reads it but TelemetryService